core/pkg/util/timeutil/timeutil.go

```diff
@@ -81,6 +83,40 @@ func DurationOffsetStrings(duration, offset time.Duration) (string, string) {
 	return DurationString(duration), DurationString(offset)
 }
 
+// ParseUTCOffset attempts to parse a UTC offset string. Returns a valid time.Duration
+// if error is nil.
+func ParseUTCOffset(offsetStr string) (time.Duration, error) {
```

Comment:
There was an implementation of this nested within the `env` utility, but we've pulled it out to allow other utilities to leverage the same parsing mechanism.